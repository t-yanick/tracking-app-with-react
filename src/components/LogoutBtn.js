import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { logIn, setUser } from '../actions/user';
import { removeAllTracks } from '../actions/tracks';
import { removeAllTrackDates } from '../actions/trackDates';

const LogoutBtn = ({
  logIn, removeAllTracks, setUser, removeAllTrackDates,
}) => {
  const onLogout = () => {
    localStorage.removeItem('token');
    logIn(false);
    setUser({});
    removeAllTracks();
    removeAllTrackDates();
  };

  return (
    <button type="button" onClick={onLogout} className="btn">Logout</button>
  );
};

const mapDispatchToProps = (dispatch) => ({
  logIn: (status) => dispatch(logIn(status)),
  setUser: (user) => dispatch(setUser(user)),
  removeAllTracks: () => dispatch(removeAllTracks()),
  removeAllTrackDates: () => dispatch(removeAllTrackDates()),
});

LogoutBtn.propTypes = {
  logIn: PropTypes.func,
  removeAllTracks: PropTypes.func,
  removeAllTrackDates: PropTypes.func,
  setUser: PropTypes.func,
};

LogoutBtn.defaultProps = {
  logIn: null,
  removeAllTracks: null,
  removeAllTrackDates: null,
  setUser: null,
};

export default connect(undefined, mapDispatchToProps)(LogoutBtn);
