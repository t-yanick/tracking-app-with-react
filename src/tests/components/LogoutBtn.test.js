import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import LogoutBtn from '../../components/LogoutBtn';
import createTestStore from '../fixtures/createTestStore';

describe('LogoutBtn', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <LogoutBtn />
          </MemoryRouter>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
