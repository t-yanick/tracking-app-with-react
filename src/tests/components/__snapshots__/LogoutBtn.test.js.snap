// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutBtn should match with snapshot 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  Logout
</button>
`;
